### PR TITLE
Fixing NPE in package naming

### LINKED ISSUE #297: NPE on package name fixing
```
package cqfn

class A
```

causes NPE

### PR DESCRIPTION
**Pull request by akuleshov7**
### What's done:
1) Fixed issue when initial package name contain only one word
2) Added 2 String rules to codestyle

Closes #297

### CODE REVIEW

File: info/diktat-kotlin-coding-style-guide-en.md [hunk 1]
```diff
@@ -1312,6 +1312,36 @@ val myStr = "Super string"
 val value = "$myStr concatenated"
 ```
 
+### <a name="r3.8"></a>Rule 3.14.2: String template format
```
Comment (akuleshov7) [inline info/diktat-kotlin-coding-style-guide-en.md:1315]: https://github.com/cqfn/diKTat/issues/301

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/PackageNaming.kt [hunk 2]
```diff
@@ -113,48 +100,44 @@ class PackageNaming(private val configRules: List<RulesConfig>) : Rule("package-
...
                 }
-            }
 
         // package name should start from a company's domain name
         if (!isDomainMatches(wordsInPackageName)) {
             PACKAGE_NAME_INCORRECT_PREFIX.warnAndFix(configRules, emitWarn, isFixMode, domainName, wordsInPackageName[0].startOffset, wordsInPackageName[0]) {
-                val parentNodeToInsert = wordsInPackageName[0].parent(DOT_QUALIFIED_EXPRESSION)!!
-                createAndInsertPackageName(parentNodeToInsert, wordsInPackageName[0].treeParent, domainName.split(PACKAGE_SEPARATOR))
+                val oldPackageName = wordsInPackageName.map { it.text }.joinToString(PACKAGE_SEPARATOR)
+                val newPackageName = "$domainName.$oldPackageName"
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/PackageNaming.kt:132]:
```suggestion
                val newPackageName = "$domainName$PACKAGE_SEPARATOR$oldPackageName"
```
to be consistent

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/PackageNaming.kt [hunk 3]
```diff
@@ -209,50 +192,38 @@ class PackageNaming(private val configRules: List<RulesConfig>) : Rule("package-
...
-                val parentNode = packageNameParts[0]
-                    .parent(PACKAGE_DIRECTIVE)!!
-                    .findChildByType(DOT_QUALIFIED_EXPRESSION)!!
+    private fun checkFilePathMatchesWithPackageName(packageNameParts: List<ASTNode>, realName: List<String>, packageDirective: ASTNode) {
 
-                parentNode.getChildren(null).forEach { node -> parentNode.removeChild(node) }
-                createAndInsertPackageName(parentNode, null, realName)
+        val realPackageNameStr = realName.joinToString(PACKAGE_SEPARATOR)
+        val offset = packageNameParts[0].startOffset
+        if (realName.isNotEmpty() && packageNameParts.map { node -> node.text } != realName) {
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/PackageNaming.kt:224]:
```suggestion
        if (realName.isNotEmpty() && packageNameParts.map { node -> node.text } != realName) {
            val realPackageNameStr = realName.joinToString(PACKAGE_SEPARATOR)
            val offset = packageNameParts[0].startOffset
```
to be more consistent with our code style recommendations on local variables declaration

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/PackageNaming.kt [hunk 4]
```diff
@@ -209,50 +192,38 @@ class PackageNaming(private val configRules: List<RulesConfig>) : Rule("package-
...
-        if (realName.isNotEmpty() && packageNameParts.map { node -> node.text } != realName) {
-            PACKAGE_NAME_INCORRECT_PATH.warnAndFix(configRules, emitWarn, isFixMode, realName.joinToString(PACKAGE_SEPARATOR),
-                    packageNameParts[0].startOffset, packageNameParts[0]) {
-                // need to get first top-level DOT-QUALIFIED-EXPRESSION
-                // -- PACKAGE_DIRECTIVE
-                //    -- DOT_QUALIFIED_EXPRESSION
-                val parentNode = packageNameParts[0]
-                    .parent(PACKAGE_DIRECTIVE)!!
-                    .findChildByType(DOT_QUALIFIED_EXPRESSION)!!
+    private fun checkFilePathMatchesWithPackageName(packageNameParts: List<ASTNode>, realName: List<String>, packageDirective: ASTNode) {
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/PackageNaming.kt:220]:
```suggestion
    private fun checkFilePathMatchesWithPackageName(packageNameParts: List<ASTNode>, realNameParts: List<String>, packageDirective: ASTNode) {
```

**Review by petertrr — APPROVED** [review]
LGTM

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/PackageNaming.kt (hunk 2 above)
Comment (akuleshov7) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/PackageNaming.kt:132]: yes true, thanks

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/PackageNaming.kt (hunk 4 above)
Comment (akuleshov7) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/PackageNaming.kt:220]: not so long line :)

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/PackageNaming.kt (hunk 3 above)
Comment (akuleshov7) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/PackageNaming.kt:224]: that's true, that should be checked by diktat 👍

**Pull request merged**